### ISSUE TITLE
Error signing assembly -- The parameter is incorrect.

### ISSUE THREAD

**Issue by mythallar** [dotnet/orleans#1117]
Hi,

I am unable to compile "Orleans Grain Interface Collection" project when "Sign the assembly" option is enabled in VS and key is password protected. I'm getting following error:

    The command ""C:\Orleans\HelloWorld\packages\Microsoft.Orleans.Templates.Interfaces.1.0.10\build\..\tools\ClientGenerator.exe" "@obj\Debug\HelloWorld.Api.codegen.args.txt"" exited with code 3.	HelloWorld.Api	

    Cryptographic failure while signing assembly 'c:\Orleans\HelloWorld\HelloWorld.Api\obj\Debug\HelloWorld.Api.dll' -- 'Error signing assembly -- The parameter is incorrect. '	HelloWorld.Api	C:\Orleans\HelloWorld\HelloWorld.Api\EXEC	
	
I'm using VS templates for Project Orleans (nuget package 1.0.10), without SDK installed.

**Comment by sergeybykov** [comment]
I just tried and was able to successfully created signed grain interface and class assemblies with Orleans 1.0.10 NuGets. Can you share you code for a repro?

**Comment by mythallar** [comment]
Source code: https://github.com/mythallar/orleans_hello_world.git
Password for key.pfx = abc123

Full MSBuild log for Orleans task:
```
1>Task "Exec"
1>  "C:\Orleans\HelloWorld\packages\Microsoft.Orleans.Templates.Interfaces.1.0.10\build\..\tools\ClientGenerator.exe" "@obj\Debug\HelloWorld.Api.codegen.args.txt"
1>  Orleans-CodeGen - command-line = "C:\Orleans\HelloWorld\packages\Microsoft.Orleans.Templates.Interfaces.1.0.10\build\..\tools\ClientGenerator.exe"  "@obj\Debug\HelloWorld.Api.codegen.args.txt"
1>  Orleans-CodeGen - Reading code-gen params from file=obj\Debug\HelloWorld.Api.codegen.args.txt
1>  MISSING: Path not exists: obj\Debug\HelloWorld.Api.dll
1>  F# compiler path not set.
1>  Orleans-CodeGen - Options 
1>  	InputLib=C:\Orleans\HelloWorld\HelloWorld.Api\obj\Debug\HelloWorld.Api.dll 
1>  	SigningKey=C:\Orleans\HelloWorld\HelloWorld.Api\key.pfx 
1>  	ServerGen=False 
1>  	CodeGenFile=C:\Orleans\HelloWorld\HelloWorld.Api\Properties\orleans.codegen.cs
1>  Orleans-CodeGen - Using referenced libraries:
1>  	Microsoft.CSharp.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\Microsoft.CSharp.dll
1>  	mscorlib.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\mscorlib.dll
1>  	Orleans.dll => C:\Orleans\HelloWorld\packages\Microsoft.Orleans.Core.1.0.10\lib\net45\Orleans.dll
1>  	System.Core.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Core.dll
1>  	System.Data.DataSetExtensions.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Data.DataSetExtensions.dll
1>  	System.Data.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Data.dll
1>  	System.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.dll
1>  	System.Xml.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Xml.dll
1>  	System.Xml.Linq.dll => C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5\System.Xml.Linq.dll
1>  Orleans-CodeGen - Generating assembly for preprocessing.
1>EXEC : error CS1548: Cryptographic failure while signing assembly 'c:\Orleans\HelloWorld\HelloWorld.Api\obj\Debug\HelloWorld.Api.dll' -- 'Error signing assembly -- The parameter is incorrect. '
1>Error : Orleans code generator found error : -- Code-gen FAILED --
1>C:\Orleans\HelloWorld\HelloWorld.Api\Exception = System.Exception : error : ClientGenerator could not compile and generate CSharp -- encountered 1 compilation warnings/errors.
1>  ErrorList = 
1>   Line 0,0 - ERROR CS1548 -- Cryptographic failure while signing assembly 'c:\Orleans\HelloWorld\HelloWorld.Api\obj\Debug\HelloWorld.Api.dll' -- 'Error signing assembly -- The parameter is incorrect. '
1>  
1>     at Orleans.CodeGeneration.GrainClientGenerator.BuildInputAssembly(CodeGenOptions options)
1>     at Orleans.CodeGeneration.GrainClientGenerator.RunMain(String[] args)
1>  
1>C:\Orleans\HelloWorld\packages\Microsoft.Orleans.Templates.Interfaces.1.0.10\build\Microsoft.Orleans.Templates.Interfaces.targets(107,5): error MSB3073: The command ""C:\Orleans\HelloWorld\packages\Microsoft.Orleans.Templates.Interfaces.1.0.10\build\..\tools\ClientGenerator.exe" "@obj\Debug\HelloWorld.Api.codegen.args.txt"" exited with code 3.
1>Done executing task "Exec" -- FAILED.
1>Done building target "OrleansClientPreprocessing" in project "HelloWorld.Api.csproj" -- FAILED.
```

I guess the suspicious line is:
```
MISSING: Path not exists: obj\Debug\HelloWorld.Api.dll
```

I can successfully compile if I disable assembly signing or remove orelans codegen targets in csproj.

**Comment by sergeybykov** [comment]
Looks like it's the password that's breaking it. If I use an .snk file as a key instead, everything works fine.

Regardless, I don't think we actually need the signing functionality in the build targets anymore.

If I manually remove $(SnkFile) related stuff from packages\Microsoft.Orleans.Templates.Interfaces.1.0.10\build\Microsoft.Orleans.Templates.Interfaces.targets, it builds fine with the password protected .pfx key.

**Comment by sergeybykov** [comment]
I see that @jthelin already mentioned the same in https://github.com/dotnet/orleans/issues/1042.

**Comment by sergeybykov** [comment]
This is already resolved in master and 1.1.0.
